source/common/runtime/runtime_impl.cc

```diff
@@ -282,16 +351,23 @@ RuntimeStats LoaderImpl::generateStats(Stats::Store& store) {
   return stats;
 }
 
-void LoaderImpl::onSymlinkSwap() {
-  current_snapshot_.reset(
-      new SnapshotImpl(root_path_, override_path_, stats_, generator_, *os_sys_calls_));
-  ThreadLocal::ThreadLocalObjectSharedPtr ptr_copy = current_snapshot_;
-  tls_->set([ptr_copy](Event::Dispatcher&) -> ThreadLocal::ThreadLocalObjectSharedPtr {
-    return ptr_copy;
-  });
+std::unique_ptr<SnapshotImpl> DiskBackedLoaderImpl::createNewSnapshot() {
+  std::vector<Snapshot::OverrideLayerSharedPtr> layers;
+  try {
+    layers.push_back(std::make_unique<DiskLayer>("root", root_path_, *os_sys_calls_));
+    if (Filesystem::directoryExists(override_path_)) {
+      layers.push_back(std::make_unique<DiskLayer>("override", override_path_, *os_sys_calls_));
+      stats_.override_dir_exists_.inc();
+    } else {
+      stats_.override_dir_not_exists_.inc();
+    }
+  } catch (EnvoyException& e) {
+    stats_.load_error_.inc();
```

Comment:
What was the behavior before when we had a load error. Did we have any runtime values at all? Just wondering if this is a change in behavior or not. It seems reasonable to have admin layer, or even root, if some layer doesn't load, but just want to be clear of what behavior was before and now.

Reply:
Before this PR, no values at all. And yes, the admin layer will be put in place even if one or both of the disk layers doesn't load. Now that I think about it, should it be all or nothing for the disk layers? What behavior do you think is best?

Reply:
I think it might be simplest to just do all or nothing for all layers. If not that, I would definitely do all or nothing for the disk layers IMO. I'm fine either way, but I think I would avoid the incremental behavior we have now.